### ISSUE TITLE
Doesn't work with redirects.

### ISSUE THREAD

**Issue by Timmmm** [donovanhide/eventsource#12]
It seems like redirects eventually lead to this code in Go's http package (`client.go`):

	func (c *Client) doFollowingRedirects(ireq *Request, shouldRedirect func(int) bool) (resp *Response, err error) {

		// ...
			if redirect != 0 {
				nreq := new(Request)
				nreq.Method = ireq.Method
				if ireq.Method == "POST" || ireq.Method == "PUT" {
					nreq.Method = "GET"
				}
				nreq.Header = make(Header)
				nreq.URL, err = base.Parse(urlStr)
				if err != nil {
					break
				}
				if len(via) > 0 {
					// Add the Referer header.
					lastReq := via[len(via)-1]
					if ref := refererForURL(lastReq.URL, nreq.URL); ref != "" {
						nreq.Header.Set("Referer", ref)
					}

					err = redirectChecker(nreq, via)
					if err != nil {
						redirectFailed = true
						break
					}
				}
				reqmu.Lock()
				req = nreq
				reqmu.Unlock()
			}

			urlStr = req.URL.String()
			if resp, err = c.send(req); err != nil {
				if wasCanceled() {
					err = &httpError{
						err:     err.Error() + " (Client.Timeout exceeded while awaiting headers)",
						timeout: true,
					}
				}
				break
			}

Basically it means that the `Accept: text/content-stream` header is lost. I don't know how to fix this but it means that redirects don't work at all. In my case I get an EOF error because the server closes the connection. This is probably a bug in Go, but can it be worked around?

**Comment by donovanhide** [comment]
Can you post a short reproduction of what you are doing to create the issue? Thanks!

**Comment by Timmmm** [comment]
It's basically this:

	stream, err := es.Subscribe("https://example.com/", "")

And then wait for events:

	for {
		select {
		case e := <-stream.Events:

The server redirects to a different address, but the headers are lost. The redirection is handled within Go's http library so it isn't easy to fix by patching eventsource. Might need to handle redirects manually but that is a huge pain!

**Comment by Timmmm** [comment]
Aha I fixed it! Pull request in a sec...

**Comment by donovanhide** [comment]
Fixed in #14
